### PR TITLE
One (last) batch of fixes. 

### PR DESCRIPTION
**Pull request by hhugo**
Sorry, I didn't bother creating a PR for each of them.

### CODE REVIEW

File: src/Fmt_ast.ml [hunk 1]
```diff
@@ -1587,7 +1596,7 @@ and fmt_expression c ?(box= true) ?epi ?eol ?parens ?ext ({ast= exp} as xexp)
                       ?ext:(if first then ext else None)
                       ctx binding
                       ~in_:(fun indent ->
-                        fmt_if_k last (break 1 (-indent) $ fmt "in") )
+                        fmt_if_k last (break 1 (~-indent) $ fmt "in") )
```
Comment (jberdine) [inline src/Fmt_ast.ml]: This doesn't look right, intended?

Comment (hhugo) [reply src/Fmt_ast.ml]: fixed

**Comment by jberdine** [comment]
Could you add a test or explanation of why you need the `Fix: keep ~+ ~-` commit at all?

**Comment by hhugo** [comment]
The part about `inner_need_parens` is needed because 
`- f x` and `-(f x)` parse differently.  

Also, `+2` and `~+2` parse_differently.

File: src/Normalize.ml [hunk 2]
```diff
@@ -70,27 +70,6 @@ let make_mapper ~ignore_doc_comment =
...
-    (* convert [~-] ident to [-] *)
-    | Pexp_apply
-        ( ( { pexp_desc= Pexp_ident ({txt= Lident "~-"} as lid)
-            ; pexp_loc
-            ; pexp_attributes } as e1 )
-        , e1N ) ->
-        m.expr m
-          (Exp.apply ~loc:pexp_loc ~attrs:pexp_attributes
-             {e1 with pexp_desc= Pexp_ident {lid with txt= Lident "-"}}
-             e1N)
```
Comment (jberdine) [inline src/Normalize.ml]: It seems that this normalization is no longer needed.

File: src/Ast.ml [hunk 3]
```diff
@@ -1075,9 +1082,14 @@ end = struct
...
        |Pexp_variant (_, Some _) ->
           Some (Apply, Non)
-      | Pexp_apply ({pexp_desc= Pexp_ident {txt= Lident i}}, [_]) -> (
+      | Pexp_apply ({pexp_desc= Pexp_ident {txt= Lident i; loc}}, [_]) -> (
         match i with
-        | "~-" | "~+" -> Some (UMinus, Non)
+        | "~-" | "~-." | "~+" | "~+." ->
+            if
+              loc.loc_end.pos_cnum - loc.loc_start.pos_cnum
+              = String.length i - 1
```
[the remark this answers is not in the document]
Comment (hhugo) [reply src/Ast.ml]: I'd like to keep source in sync with the ocaml compiler.
I'll add a test.

**Comment by hhugo** [comment]
That works for me.

File: src/Ast.ml (hunk 3 above)
[the remark this answers is not in the document]
Comment (hhugo) [reply src/Ast.ml]: done

**Pull request merged**